mbed_build/_internal/config/config.py

```diff
@@ -105,7 +106,9 @@ def from_sources(cls, sources: Iterable[Source]) -> "Config":
         for source in sources:
             for key, value in source.config.items():
                 _create_config_option(config, key, value, source)
-            for key, value in source.config_overrides.items():
+            for key, value in source.overrides.items():
+                if key in ALL_CUMULATIVE_FIELDS:
```

Comment:
I find this a little tricky to reason about. Why are the cumulative attributes ignored here now? Are we expecting that they will have already made any modifications?